src/main/java/strategy/OperationStrategyImpl.java

```
@@ -0,0 +1,17 @@
+package strategy;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> handlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> handlers) {
+        this.handlers = handlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
+        return handlers.get(operation);
```

getHandler currently returns `handlers.get(operation)` which may return null if no handler is registered. Instead the strategy should fail fast and provide an informative error. Change implementation to validate input and throw a RuntimeException like: `if (operation == null) throw new RuntimeException("Operation is null"); OperationHandler handler = handlers.get(operation); if (handler == null) throw new RuntimeException("No handler registered for operation: " + operation); return handler;`. This prevents a later NullPointerException in callers and matches the expected Strategy behaviour in the task flow (see how the strategy is used from Main/ShopService)  .